linera-core/src/client/mod.rs

```diff
@@ -1213,7 +1211,12 @@ where
             let sender_chain_id = certificate.value().chain_id();
             let height = certificate.value().height();
             let epoch = certificate.value().epoch();
-            match self.check_certificate(max_epoch, &committees, &certificate)? {
+            debug_assert!(certificate.value().is_confirmed());
```

Comment:
This could actually happen, if the remote node is faulty?

I'd actually put this into the `RemoteNode::download_certificates` implementation: If they send us the wrong type, return a `NodeError`. Otherwise return a list of `ConfirmedBlockCertificate`s.

Reply:
So validators are never expected to return a `ValidatedBlockCertificate` from `download_certificates`?

Reply:
The problem with changing `download_certificates` logic for `RemoteNode` is that the `ValidatorNode` trait is also implemented for client where we might not expect _only_ `ConfirmedBlockCertificate`.

Also, this change would change the RPC schema which I tried to avoid.

Reply:
I changed this to return an error _here_ if the `certificate` is not of `ConfirmedBlock` variant.

Reply:
I don't think so. IIRC the validators look up the certificates in storage, and we only store confirmed blocks.

(OK, we also store validated blocks, timeouts and proposals, but only as part of the chain manager.)

Reply:
(Anyway, we can revisit this as part of the later PR that breaks backwards compatibility of the protocol anyway.)